### PR TITLE
DBAAS-257: add docs teams' RHODA quick-starts on installation

### PR DESCRIPTION
**Pull request by jeremyary**
Adds quickstart content provided by our docs teams to the cluster by means of a new installation phase. The content comes from here. The result is that if you access Quick Starts, either by the link from our OLM operator card or via the "?" help menu in the top-right corner, you will be able to access these new Quick Start cards:

### CODE REVIEW

File: controllers/reconcilers/quickstart_installation/quickstart_installation_reconciler.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+	"github.com/RHEcosystemAppEng/dbaas-operator/api/v1alpha1"
+	"github.com/RHEcosystemAppEng/dbaas-operator/controllers/reconcilers"
+)
+
+//go:embed creating-a-provider-account-instance-quick-start.yaml
+var providerAccountQuickStart []byte
+
+//go:embed connecting-to-an-external-database-instance-quick-start.yaml
+var connectionQuickStart []byte
```
Comment (jeremyary) [inline controllers/reconcilers/quickstart_installation/quickstart_installation_reconciler.go:24]: to point out the new voodoo here, these lines grab YAML files in the same directory & shove them into byte slices, simple as that. I don't touch them again until they `yaml.unmarshal` @L58

File: controllers/dbaasplatform_controller.go [hunk 2]
```diff
@@ -239,6 +240,7 @@ func (r *DBaaSPlatformReconciler) createPlatformCR(ctx context.Context, serverCl
 func (r *DBaaSPlatformReconciler) getInstallationPlatforms() []dbaasv1alpha1.PlatformsName {
 
 	return []dbaasv1alpha1.PlatformsName{
+		dbaasv1alpha1.DBaaSQuickStartInstallation,
```
Comment (redhatHameed) [inline controllers/dbaasplatform_controller.go:243]: can we change the order of quick start installation, adding in the last? WDYT?

**Comment by redhatHameed** [comment]
> Adds quickstart content provided by our docs teams to the cluster by means of a new installation phase. The content comes from here. The result is that if you access Quick Starts, either by the link from our OLM operator card or via the "?" help menu in the top-right corner, you will be able to access these new Quick Start cards:
> 
> 
> 
> 
> 
> 

are we planning to add creating the trial instance in a quick start doc?

**Comment by jeremyary** [comment]
We will have another card, but it's up to Docs as to what cards we get or how they're combined. We'll just receive YAML and put into directory. The provisioning card will be ready once we can get docs team solid UI and e2e experience so they can finish it up.

File: controllers/dbaasplatform_controller.go (hunk 2 above)
Comment (jeremyary) [reply controllers/dbaasplatform_controller.go]: sure, updated

**Review by redhatHameed — APPROVED** [review]
LGTM

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: controllers/reconcilers/quickstart_installation/quickstart_installation_reconciler.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			return status, err
+		}
+	}
+	return v1alpha1.ResultSuccess, nil
+}
+
+func (r *Reconciler) createQuickStartCR(qsName string, qsBytes []byte, ctx context.Context) (v1alpha1.PlatformsInstlnStatus, error) {
+	quickStart := r.getQuickStartModel(qsName)
+	quickStartFromFile := &consolev1.ConsoleQuickStart{}
+	err := yaml.Unmarshal(qsBytes, quickStartFromFile)
```
[the remark this answers is not in the document]
Comment (jeremyary) [reply controllers/reconcilers/quickstart_installation/quickstart_installation_reconciler.go:58]: yeah that's fair, I wasn't sure how much resource creation was worth testing but I adding cvg for embed/marshalling makes sense. Update shortly.

[the remark this answers is not in the document]
Comment (jeremyary) [reply controllers/reconcilers/quickstart_installation/quickstart_installation_reconciler.go:58]: tests added to unmarshal each ConsoleQuickStart YAML and make several assertions about their content.

**Review by tchughesiv — APPROVED** [review]
lgtm

**Pull request merged**